src/api/factories/ProposalFactory.ts

```diff
@@ -81,7 +81,22 @@ export class ProposalFactory {
      * @param {ProposalMessage} proposalMessage
      * @returns {Promise<ProposalCreateRequest>}
      */
-    public async getModel(proposalMessage: ProposalMessage, smsgMessage: resources.SmsgMessage): Promise<ProposalCreateRequest> {
+    public async getModel(proposalMessage: ProposalMessage, smsgMessage: resources.SmsgMessage | null): Promise<ProposalCreateRequest> {
```

Comment:
avoid using null's and instead make the smsgMessage optional.
public async getModel(proposalMessage: ProposalMessage, smsgMessage?: resources.SmsgMessage)